src/media/local-stream.ts

```diff
@@ -148,11 +149,8 @@ abstract class _LocalStream extends Stream {
    */
   async addEffect(effect: TrackEffect): Promise<void> {
     // Check if the effect has already been added.
-    if (this.effects.includes(effect)) {
-      throw new WebrtcCoreError(
-        WebrtcCoreErrorType.ADD_EFFECT_FAILED,
-        `Effect ${effect.id} has already been added to this stream.`
-      );
+    if (this.effects.some((e) => e.id === effect.id)) {
```

Comment:
So this now checks if the _same_ effect instance is being added, but I think we need to check for an existing effect of the same kind to replace it with this one?

Reply:
Oh, I didn't notice that all the code below is in this same method, I see there's handling there  for this.

Reply:
There's logic starting at line 205 that handles this case.